angucomplete-alt.js

```diff
@@ -217,20 +217,43 @@
       }
 
       function findMatchString(target, str) {
+
         var result, matches, re;
-        // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
-        // Escape user input to be treated as a literal string within a regular expression
-        re = new RegExp(str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'), 'i');
+
         if (!target) { return; }
         if (!target.match || !target.replace) { target = target.toString(); }
-        matches = target.match(re);
-        if (matches) {
-          result = target.replace(re,
-              '<span class="'+ scope.matchClass +'">'+ matches[0] +'</span>');
+
+        if (scope.matchAllTokens) {
+
+          // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
+          // Escape user input to be treated as a literal string within a regular expression
+          re = new RegExp('('+str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&').match(/\S+/g).join('|')+')', 'gi');
```

Comment:
It would be nice to explain in a codecomment what happens here - since its hard to follow due to string concatenation, replace, matching and joining